onnxruntime/core/session/inference_session.cc

```diff
@@ -810,6 +810,20 @@ std::pair<common::Status, const InputDefList*> InferenceSession::GetModelInputs(
   return std::make_pair(common::Status::OK(), &model_->MainGraph().GetInputs());
 }
 
+std::pair<common::Status, const InputDefList*> InferenceSession::GetModelInputsIncludingInitializers() const {
+  {
+    std::lock_guard<onnxruntime::OrtMutex> l(session_mutex_);
+    if (!is_model_loaded_) {
+      LOGS(*session_logger_, ERROR) << "Model was not loaded";
+      return std::make_pair(common::Status(common::ONNXRUNTIME, common::FAIL, "Model was not loaded."),
+                            nullptr);
+    }
+  }
+
+  // return required inputs including any inputs used for overriding initializers
+  return std::make_pair(common::Status::OK(), &model_->MainGraph().GetInputsIncludingInitializers());
+}
```

Comment:
if graph.CanOverrideInitializer() is false this should only return the values from GetInputs().

Reply:
Or we should consider updating the logic in graph.cc to only have overrideable initializers in this list (IR version >= 4 and initializer has matching graph input). That's probably cleaner but need to check if any partner relies on the current behaviour.

---
In reply to: [324982805](https://github.com/microsoft/onnxruntime/pull/1854#discussion_r324982805) [](ancestors = 324982805)